ui/de-lib/src/main/java/org/iplantc/de/client/models/sharing/Sharing.java

```diff
@@ -9,29 +14,67 @@
  */
 public class Sharing {
```

Comment:
If you are going to use a class like this, it would be better to implement as a [Javascript Overlay Type](http://www.gwtproject.org/doc/latest/DevGuideCodingBasicsOverlay.html). It's not very clear in that documentation how to serialize the json into the overlay type, but you can find a good example in the _"Working with JSON"_ section, [Parsing JSON](http://www.gwtproject.org/doc/latest/DevGuideCodingBasicsJSON.html#parsing).